src/Util/Profiler.php

```diff
@@ -69,21 +69,27 @@ public function isRendertime(): bool
 	/**
 	 * Updates the enabling of the current profiler
 	 *
+	 * Note: The reason there are two different ways of updating the configuration of this class is because it can
+	 * be used even with no available database connection which IManageConfigValues doesn't ensure.
+	 *
 	 * @param IManageConfigValues $config
 	 */
 	public function update(IManageConfigValues $config)
 	{
-		$this->enabled = $config->get('system', 'profiler');
-		$this->rendertime = $config->get('rendertime', 'callstack');
+		$this->enabled = (bool) $config->get('system', 'profiler') ?? false;
```

Comment:
Shouldn't this be done this way: `$config->get('system', 'profiler', false)`?

Reply:
Two answers:
1. In this method, yes, because it's a `IManageConfigValues` object, but we still need to cast to bool afterwards because the value from the config is arbitrary. In the constructor, the `ConfigCache->get()` method doesn't accept a default value, so I preferred having the same call in both places. I added documentation to distinguish both usages.
2. Yes for this specific config value, but we can't provide a default value for `rendertime.callstack` since it's an addon config. If you include the rendertime in the core (which it should be since it's referenced in the core), we'll be able to provide a default value in `static/defaults.config.php`.